src/main/java/org/breedinginsight/brapi/v2/services/BrAPIObservationVariableService.java

```diff
@@ -55,6 +56,8 @@ public BrAPIObservationVariableService(
         this.referenceSource = referenceSource;
     }
 
+    // TODO: support sub-entity datasets.
+    // This gets the observation variables for the top-level dataset in an experiment.
     public List<Trait> getBrAPIObservationVariablesForExperiment(
```

Comment:
It looks like this is only being called via the brapi endpoint controller so it doesn't impact the front-end does it?

Reply:
I think this method will eventually need to accept a `datasetId` parameter, so that it can be used for a specific dataset. It's outside the scope of this story because BI-2109 doesn't include any means of creating Observation/ObservationVariables for sub-entity datasets.